code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -203,6 +206,30 @@
 	return RM
 
 /obj/item/robot_module/proc/be_transformed_to(obj/item/robot_module/old_module)
+	if(islist(borg_skins))
+		var/mob/living/silicon/robot/R = loc
+		var/list/reskin_icons = list()
+		for(var/skin in borg_skins)
+			var/list/details = borg_skins[skin]
+			reskin_icons[skin] = image(icon = details[SKIN_ICON] || 'icons/mob/robots.dmi', icon_state = details[SKIN_ICON_STATE])
+		var/borg_skin = show_radial_menu(R, R, reskin_icons, custom_check = CALLBACK(src, .proc/check_menu, R, old_module), radius = 38, require_near = TRUE)
+		if(!borg_skin)
+			return FALSE
+		var/list/details = borg_skins[borg_skin]
+		if(!isnull(details[SKIN_ICON_STATE]))
+			cyborg_base_icon = details[SKIN_ICON_STATE]
+		if(!isnull(details[SKIN_ICON]))
+			R.icon = details[SKIN_ICON]
+		if(!isnull(details[SKIN_PIXEL_X]))
+			R.pixel_x = details[SKIN_PIXEL_X]
+		if(!isnull(details[SKIN_PIXEL_Y]))
+			R.pixel_y = details[SKIN_PIXEL_Y]
+		if(!isnull(details[SKIN_LIGHT_KEY]))
+			special_light_key = details[SKIN_LIGHT_KEY]
+		if(!isnull(details[SKIN_HAT_OFFSET]))
+			hat_offset = details[SKIN_HAT_OFFSET]
+		if(!isnull(details[SKIN_TRANSFORM]))
+			animated_transform = details[SKIN_TRANSFORM]
```

Comment:
do you need this part? If you don't assign any details they'll be null as well

Reply:
Well it checks if it specifically isn't null. Just assigning values to null can be destructive in some cases, like when assigning icon or an icon state. It's probably not that needed but a bit of safety doesn't hurt.

Reply:
use ||= instead.

halves the lines required.

A ||= B will set A to B only if A is null.

Reply:
> do you need this part? If you don't assign any details they'll be null as well

yes, assigning icon_state to null is bad